47.md

```diff
@@ -373,6 +373,33 @@ Response:
 }
 ```
 
+### `get_budget`
+
+Budgets can be used to limit the amount of funds that can be spent by a connection in a given time period.
+Budgets are optional and can be set by the user if their wallet supports it. Support for this command indicates
+that the wallet supports user-configured budgets. The budget will be reset at the `renews_at` timestamp.
+
+Request:
+```jsonc
+{
+    "method": "get_budget",
+    "params": {
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "get_budget",
+    "result": {
+        "remaining_budget_msats": 10000,
```

Comment:
Instead of returning the remaining budget, I suggest returning the used budget. I reason that when a user changes the total budget, the remaining budget will be affected, while the used budget will not change. Because of this, I imagine most developers will track a used amount to subtract from the total amount to return, so instead of forcing everyone to do that logic, it seems to me it would make more sense to return the used budget amount.

Reply:
Interesting, yeah I suppose that makes sense too. I see your point about the total budget potentially changing and moving the remaining amount. I guess I'd been thinking that apps would usually want to know how much they have left to spend more often than how much they've already spent (although of course it's an easy calculation either way). I'm generally not totally opposed to the switch, but would love more opinions on this. 

cc @rolznz @bumi - curious if you've seen these behavior patterns either way in nwc apps.

Reply:
If it stays as the remaining balance, the amount should be a signed integer to limit any potential incompatibility issues.

The values may be unsigned integers if it is changed to the used balance.

Reply:
I think I would prefer to display the `used` budget here too for the reasons @davidcaseria mentioned.

Reply:
Sounds good, will swap.